static/js/apps/biomed_nl/app.tsx

```diff
@@ -86,23 +102,125 @@ export function App(): ReactElement {
   const [answer, setAnswer] = useState<DisplayedAnswer>(null);
   const [showLoading, setShowLoading] = useState<boolean>(false);
 
+  // A loading spinner that shows new biomed pun every minute
+  function SpinnerWithText(): ReactElement {
+    const [showMessage, setShowMessage] = useState(false);
+    const [message, setMessage] = useState("");
+    const [messageIndex, setMessageIndex] = useState(0);
+
+    const customMessages = useMemo(
+      () => [
+        "The data is taking its time to ensure perfect homeostasis.",
+        "Like a healthy cell, the data is going through a natural maturation process.",
+        "The data is undergoing a controlled differentiation, becoming more specialized.",
+        "The data is percolating through the system, ensuring complete absorption.",
+        "The data is taking its time to allow for complete osmosis of information.",
+        "The data is making sure all its enzymes are perfectly aligned for the reaction.",
+      ],
+      []
+    );
+    // Show puns in random order every time.
+    const shuffledIndexes = useRef(
+      [...Array(customMessages.length).keys()].sort(() => Math.random() - 0.5)
+    ).current;
+
+    useEffect(() => {
+      // Show first biomed pun after 3 seconds
+      const timeoutId = setTimeout(() => {
+        setShowMessage(true);
+        setMessage(customMessages[shuffledIndexes[messageIndex]]);
+      }, 3000);
+
+      // Show next pun after one minute
```

Comment:
do you mean "show new pun every minute"? because I believe that's what your code is doing

Reply:
yep! clarified the comment